src/store_provider.rs

```diff
@@ -6,7 +6,7 @@ use object_store::path::Path;
 use object_store::ObjectStore;
 use std::sync::Arc;
 
-pub(crate) trait StoreProvider {
+pub(crate) trait StoreProvider: Send + Sync {
```

Comment:
wrap a `#[async_trait]` here? or the type system may screams if we want to use `Box<dyn StoreProvider>`.

related: #539